File: WorkBranch/backend/controller/VO/result.py
class Result:
    def __init__(self, code: int, message: str = "", data: object = None):
        self.code = code
        self.message = message
        self.data = data or None

    def success(self,code:int = 200, message: str = "Success", data: object = None) -> 'Result':
        self = self.__init__(code, message, data)
        return self

    def error(self, code: int = 500, message: str = "Error", data: object = None) -> 'Result':
        self = self.__init__(code, message, data)
        return self

File: packages/backend/src/controller/result.ts
export interface Result<T = unknown> {
  code: number;
  message: string;
  data: T | null;
}

export function success<T>(data: T | null = null): Result<T> {
  return {
    code: 0,
    message: 'success',
    data,
  };
}

export function error(message: string, code: number = 500): Result {
  return {
    code,
    message,
    data: null,
  };
}
